packages/server/src/adapters/fetch/fetchRequestHandler.ts

```diff
@@ -0,0 +1,62 @@
+import { assertNotBrowser } from '../../assertNotBrowser';
+import { HTTPRequest } from '../../http/internals/types';
+import { resolveHTTPResponse } from '../../http/resolveHTTPResponse';
+import { AnyRouter } from '../../router';
+import { FetchHandlerOptions } from './types';
+
+assertNotBrowser();
+
+export type FetchHandlerRequestOptions<TRouter extends AnyRouter> = {
+  req: Request;
+  endpoint: string;
+} & FetchHandlerOptions<TRouter>;
+
+export async function fetchRequestHandler<TRouter extends AnyRouter>(
+  opts: FetchHandlerRequestOptions<TRouter>,
+): Promise<Response> {
+  const createContext = async () => {
+    return opts.createContext?.({ req: opts.req });
+  };
+
+  const url = new URL(opts.req.url);
+  const path = url.pathname.slice(opts.endpoint.length + 1);
+  console.log(path);
+  const req: HTTPRequest = {
+    query: url.searchParams,
+    method: opts.req.method,
+    headers: Object.fromEntries(opts.req.headers),
+    body: await opts.req.text(),
+  };
```

Comment:
Later, we should probably invert this lower down so `fetch` is the default and that the "old stuff" are mapped instead

Reply:
Good idea, definitely